test_embeding test does not pass as the embedding values differ

Running our tests on a GPU results in an error:
Run using `pytest tests/embedding/test_embedding.py`

```
>       np.testing.assert_allclose(embeddings_1_worker, embeddings_4_worker, rtol=5e-5)
E       AssertionError: 
E       Not equal to tolerance rtol=5e-05, atol=0
E       
E       Mismatched elements: 54 / 320 (16.9%)
E       Max absolute difference: 0.00010848
E       Max relative difference: 0.00162661
E        x: array([[-3.825889e-02, -4.968233e-01, -4.616053e-01,  1.188970e+00,
E                5.971792e-01,  7.000559e-02, -2.529578e-01, -6.173887e-01,
E                2.215651e-01, -1.513715e-02, -7.346951e-01, -4.753737e-01,...
E        y: array([[-3.824084e-02, -4.968480e-01, -4.616520e-01,  1.189015e+00,
E                5.971847e-01,  7.001890e-02, -2.529720e-01, -6.173834e-01,
E                2.215403e-01, -1.511938e-02, -7.347519e-01, -4.753903e-01,...

tests/embedding/test_embedding.py:69: AssertionError
```

Running the same code on the CPU using `CUDA_VISIBLE_DEVICES= pytest tests/embedding/test_embedding.py ` works though.

--- a/tests/embedding/test_embedding.py
+++ b/tests/embedding/test_embedding.py
@@ -58,6 +58,14 @@ class TestLightlyDataset(unittest.TestCase):
         )
 
         manual_seed(43)
+        dataloader_1_worker_2 = DataLoader(
+            dataset, shuffle=True, num_workers=0, batch_size=4
+        )
+        embeddings_1_worker_2, *_ = encoder.embed(
+            dataloader_1_worker_2,
+            device=device,
+        )
+
         dataloader_4_worker = DataLoader(
             dataset, shuffle=True, num_workers=4, batch_size=4
         )
@@ -66,7 +74,10 @@ class TestLightlyDataset(unittest.TestCase):
             device=device,
         )
 
-        np.testing.assert_allclose(embeddings_1_worker, embeddings_4_worker, rtol=5e-5)
+        np.testing.assert_allclose(
+            embeddings_1_worker, embeddings_1_worker_2, rtol=1e-1
+        )
+        np.testing.assert_allclose(embeddings_1_worker, embeddings_4_worker, rtol=1e-1)
         np.testing.assert_allclose(labels_1_worker, labels_4_worker, rtol=1e-5)
 
         self.assertListEqual(filenames_1_worker, filenames_4_worker)

--- a/tests/embedding/test_embedding.py
+++ b/tests/embedding/test_embedding.py
@@ -58,14 +58,6 @@ class TestLightlyDataset(unittest.TestCase):
         )
 
         manual_seed(43)
-        dataloader_1_worker_2 = DataLoader(
-            dataset, shuffle=True, num_workers=0, batch_size=4
-        )
-        embeddings_1_worker_2, *_ = encoder.embed(
-            dataloader_1_worker_2,
-            device=device,
-        )
-
         dataloader_4_worker = DataLoader(
             dataset, shuffle=True, num_workers=4, batch_size=4
         )
@@ -74,11 +66,8 @@ class TestLightlyDataset(unittest.TestCase):
             device=device,
         )
 
-        np.testing.assert_allclose(
-            embeddings_1_worker, embeddings_1_worker_2, rtol=1e-1
-        )
-        np.testing.assert_allclose(embeddings_1_worker, embeddings_4_worker, rtol=1e-1)
-        np.testing.assert_allclose(labels_1_worker, labels_4_worker, rtol=1e-5)
+        np.testing.assert_allclose(embeddings_1_worker, embeddings_4_worker, atol=5e-4)
+        np.testing.assert_allclose(labels_1_worker, labels_4_worker, atol=1e-5)
 
         self.assertListEqual(filenames_1_worker, filenames_4_worker)
         self.assertListEqual(filenames_1_worker, dataset.get_filenames())

